src/sysl/exporters/json_out/serializer.py

```diff
@@ -88,75 +88,155 @@ def serializer(context):
         for (tname, t) in sorted(app.types.iteritems()):
             if not t.WhichOneof('type') in ['relation', 'tuple']:
                 continue
-
-            w()
-            with java.Method(w, 'public', 'void', 'serialize' + tname + 'View',
-                             [('JsonGenerator', 'g'),
-                              (tname + '.View', 'view')],
-                             throws=['IOException']):
-                with java.If(w, 'view.isEmpty()'):
-                    w('return;')
-                w('g.writeArrayFieldStart("{}");', tname)
-                with java.For(w, '{} item : view', tname):
-                    w('g.writeStartObject();')
+            java.SeparatorComment(w)
+            if t.WhichOneof('type') == 'relation':
+                w()
+                with java.Method(w, 'public', 'void', 'serialize' + tname + 'View',
+                                 [('JsonGenerator', 'g'),
+                                  (tname + '.View', 'view')],
+                                 throws=['IOException']):
+                    with java.If(w, 'view.isEmpty()'):
+                        w('return;')
+                    w('g.writeArrayFieldStart("{}");', tname)
+                    with java.For(w, '{} item : view', tname):
+                        w('g.writeStartObject();')
+                        for (fname, f) in datamodel.sorted_fields(t):
+                            jfname = java.name(fname)
+                            method = java.CamelCase(jfname)
+                            type_ = datamodel.typeref(f, module)[2]
+                            extra = '{}'
+                            which_type = type_.WhichOneof('type')
+                            if which_type == 'primitive':
+                                (jsontype,
```

Comment:
There are no parenthesis required around an expression list i.e. `jsontype, extra = ...` is enough.
There are a few instances of this in this change - please remove parenthesis in all occurrences and also in `for` statements e.g.`for x, y in xy_list:` rather than `for (x, y) in xy_list:.

Reply:
i think (x,y) is better.

Reply:
Ok, this seems to be an opinion issue. 
I've tried hard to find a reference with authority that tuple unpacking e.g. `a, b = b, a` is more pythonic than with parens but I couldn't, so let's drop this.

Reply:
I prefer consistently including the parens because it's always clear and unambiguous, whereas it isn't always obvious when parens may be safely omitted, e.g.: `[a, b for b, a in range(5)]`, and `world[x, y]` vs `world[x,y:z]`. I also don't like the inconsistency of having a different rule for the outer node of a deep structure: `for i, (x, y) in enumerate(points):`

(I am of course at odds with most Python programmers, who favour omitting them whenever possible.)